dask_expr/tests/test_concat.py

```diff
@@ -139,3 +139,23 @@ def test_concat_index(df, pdf):
     query = Len(result.expr).optimize(fuse=False)
     expected = (0 + Len(df.expr) + Len(df2.expr)).optimize(fuse=False)
     assert query._name == expected._name
+
+
+def test_concat_after_merge():
```

Comment:
I'm sure a simpler test can be used to cover this fix. This is just similar to the tpch query I was exploring when the bug showed up.